docs/infrastructure/README.md

```diff
@@ -0,0 +1,167 @@
+application; staging and production. The staging environment should be deployed upon push to the `main` branch for the
+relevant repository, and the production environment should be deployed upon a tag release to the same repository.
+
+This ensures the entire deployment process is automated, and can be done without developer involvement. Evergreen is
+a standard EC2 instance, and we will allow GitHub Actions to SSH into the instance to run the deployment steps.
+
+**Deployment Steps**
+
+1. GitHub Actions builds the Docker image and pushes it to AWS ECR.
+2. GitHub Actions accesses Evergreen via SSH
+3. Pull the Docker image from ECR
+4. Re-fetch the latest environment variables for the application from Doppler
+5. Restart the containers that are used by the application
+
+Deployment will be done through a set of bash scripts that are developed in the terraform monorepo and copied over to
+the AWS AMI that Evergreen runs on.
+
+**Re-deploying Evergreen**
+
+Each time an application is added to Evergreen, or the docker-compose.yml file is updated, the Evergreen instance must
+be re-deployed. This can be automatically done by Terraform by re-building the Packer AMI, and re-deploying the EC2
+instance with the new AMI.
+
+- Update configuration in Terraform monorepo / docker-compose.yml
+- Re-build Packer AMI
+- Deploy new EC2 instance with new AMI
+- Route CloudFront to the new EC2 instance and invalidate the entire distribution
+- Destroy old EC2 instance
+
+All applications can be served through Traefik, and should be tagged in the `docker-compose.yml` file with the necessary
+Traefik labels for Traefik to automatically pick up.
+
+The same steps should be applied when the Evergreen instance is scaled up or down. This can be done by re-deploying the
+instance with the new AMI, as the instance type is configured in the Packer template and `aws_instance` resource.
+
+## Price
+
+The application fleet should stay relatively cheap, as we run many applications on a single host in order to maximize
+usage of the instance. The price of the Evergreen instance is based on the following factors:
+
+- Instance type: t4g.medium
+- Static IPv4 address
+- EBS Storage on the instance (should be limited)
+- CloudFront (practically free)
+- Additional costs for VPC, Route53, other related services
+
+## Pitfalls & Alternatives
+
+Evergreen is a single instance, meaning there is a single point of failure for the entire application fleet. If the
+instance is down, all applications will be down. This should not be a very big problem for a student organization like
+ours. However, it is important to keep in mind that the instance is not intended to be highly scalable, and should
+likely be used for smaller applications that we don't mind small downtimes for.
+
+Whenever we want to scale up (Åre påmelding), we should deploy a new Evergreen instance.
```

Comment:
I still don't think we need scaling for åre, we just need signup to be async

Literally just introduce a `signup-state: pending`, throw a message on SQS on signup, have a lambda process signups one-by-one

Reply:
Maybe this is easy to implement for ow4? Although all service logic is coupled to django app. But we could just write sql directly in lambda maybe?